hack/nightly.sh

```diff
@@ -4,19 +4,6 @@ RELEASE_REPO=${RELEASE_REPO:-public.ecr.aws/z3c6l6z9/karpenter-nightly}
 RELEASE_VERSION=${RELEASE_VERSION:-$(date "${NIGHTLY_TAG_FMT}")}
 RELEASE_PLATFORM="--platform=linux/amd64,linux/arm64"
 
-if [ -z "$CLOUD_PROVIDER" ]; then
-    echo "CLOUD_PROVIDER environment variable is not set: 'export CLOUD_PROVIDER=aws'"
-    exit 1
-fi
```

Comment:
Why do we want to remove this? We'd want to know if no cloud provider is set so we don't silently default to a mock environment.

Reply:
this was removed from https://github.com/aws/karpenter/pull/1587 and it's not needed in the nightly as there's no mock env.

Reply:
the same thing could happen in the nightly release. If the env var isn't set, Karpenter will build with our "fake" cloud provider. IMO this should be added back to the nightly script.

Reply:
Alright added back.